server/src/main/java/com/kritsit/casetracker/server/datalayer/UserRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.kritsit.casetracker.server.datalayer;
+
+import java.util.Map;
+
+import com.kritsit.casetracker.server.domain.model.Permission;
+import com.kritsit.casetracker.server.domain.model.Staff;
+
+public class UserRepository implements IUserRepository {
+	
+	private final IPersistenceService db;
+	
+	public UserRepository(){
+		//Just until we have mockin \ DI
+		this.db = new DatabasePersistence();
+	}
+	
+	public UserRepository(IPersistenceService db){
+		this.db = db;
+	}
+	
+    public long getPasswordSaltedHash(String username) throws RowToModelParseException {
+    	try {
+	        String sql = "SELECT passwordHash FROM staff WHERE username=\'" 
+	            + username + "\';";
+	        Map<String, String> rs = db.executeQuery(sql);
+	        return Long.parseLong(rs.get("passwordHash"));
+    	}
+    	catch(Exception e){
```

Comment:
Just checked the method and no exceptions are thrown. The only time an exception is thrown is when parsing the long. NullPointerException if user isn't found or ParseException if trying to parse something that is not a long